src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/masonry-decorator/masonry-view.js

```diff
@@ -0,0 +1,414 @@
+                this.selectRecord(id);
+            }
+
+            this.sandbox.dom.append(this.sandbox.dom.find('#' + constants.masonryGridId, this.$el), this.$items[id]);
+            this.bindItemLoadingEvents(id);
+            this.bindItemDomEvents(id);
+        },
+
+        /**
+         * Binds dom-related events on a masonry-grid item
+         * @param id the identifier of the thumbnail to bind events on
+         */
+        bindItemDomEvents: function(id) {
+            this.sandbox.dom.on(this.$items[id], 'click', function(event) {
+                this.sandbox.dom.stopPropagation(event);
+                this.datagrid.itemAction.call(this.datagrid, id);
+            }.bind(this), "." + constants.actionNavigatorClass);
+
+            this.sandbox.dom.on(this.$items[id], 'click', function(event) {
+                this.sandbox.dom.stopPropagation(event);
+                this.sandbox.emit(DOWNLOAD_CLICKED.call(this), id);
+            }.bind(this), "." + constants.downloadNavigatorClass);
+
+            if (!!this.options.selectable) {
+                this.sandbox.dom.on(this.$items[id], 'click', function(event) {
+                    this.sandbox.dom.stopPropagation(event);
+                    this.toggleItemSelected(id);
+                }.bind(this));
+            }
+        },
+
+        /**
+         * Bind image-loading events on a masonry-grid item
+         * @param id
+         */
+        bindItemLoadingEvents: function(id) {
+            this.sandbox.dom.one(this.sandbox.dom.find('.' + constants.headImageClass, this.$items[id]), 'load', function() {
+                this.sandbox.dom.remove(this.sandbox.dom.find('.' + constants.headIconClass, this.$items[id]));
+                this.sandbox.masonry.refresh('#' + constants.masonryGridId, true);
+                this.sandbox.dom.removeClass(this.$items[id], constants.loadingClass);
+            }.bind(this));
+
+            this.sandbox.dom.one(this.sandbox.dom.find('.' + constants.headImageClass, this.$items[id]), 'error', function() {
+                this.sandbox.dom.remove(this.sandbox.dom.find('.' + constants.headImageClass, this.$items[id]));
+                this.sandbox.masonry.refresh('#' + constants.masonryGridId, true);
+                this.sandbox.dom.removeClass(this.$items[id], constants.loadingClass);
+            }.bind(this));
+        },
+
+        /**
+         * Toggles an item with a given id selected or unselected
+         * @param id {Number|String} the id of the item
+         */
+        toggleItemSelected: function(id) {
+            if (this.datagrid.itemIsSelected.call(this.datagrid, id) === true) {
+                this.deselectRecord(id);
+            } else {
+                this.selectRecord(id);
+            }
+        },
+
+        /**
+         * Takes an object with options and extends the current ones
+         * @param options {Object} new options to merge to the current ones
+         */
+        extendOptions: function(options) {
```

Comment:
@marcelmoosbrugger this function is used by the datagrid component on view-change